contracts/hackatom/src/contract.rs

```diff
@@ -75,13 +79,30 @@ pub fn handle<S: Storage, A: Api>(
     }
 }
 
-pub fn query<S: Storage, A: Api>(_deps: &Extern<S, A>, msg: Vec<u8>) -> Result<Vec<u8>> {
-    let _msg: QueryMsg = from_slice(&msg).context(ParseErr { kind: "QueryMsg" })?;
-    Ok(vec![])
+pub fn query<S: Storage, A: Api>(deps: &Extern<S, A>, msg: Vec<u8>) -> Result<Vec<u8>> {
+    let msg: QueryMsg = from_slice(&msg).context(ParseErr { kind: "QueryMsg" })?;
+    match msg {
+        QueryMsg::Verifier{} => query_verifier(deps),
+    }
+}
+
+fn query_verifier<S: Storage, A: Api>(deps: &Extern<S, A>) -> Result<Vec<u8>> {
+    let data = deps.storage.get(CONFIG_KEY).context(ContractErr {
+        msg: "uninitialized data",
+    })?;
+    let state: State = from_slice(&data).context(ParseErr { kind: "State" })?;
+    let addr = deps.api.human_address(&state.verifier)?;
+    // we just pass the address as raw bytes
+    // these will be base64 encoded into the json we return, and parsed on the way out.
+    // maybe we should wrap this in a struct then json encode it into a vec?
+    // other ideas?
```

Comment:
I don't see at which point this is converted to JSON at all. In order to get JSON responses, I created simple wrapper types: https://github.com/confio/cosmwasm-examples/pull/10/commits/9e011f4152611be253ddbdc0291550036cf30753

Reply:
Yes, https://github.com/confio/cosmwasm-examples/commit/9e011f4152611be253ddbdc0291550036cf30753#diff-65f2aed784f599ceef998401efc7a9ecR159-R165 is a good example how to do this in a real contract.

I just use a raw string for simple case, to ensure the ok and error paths work.

Reply:
I don't really understand the comment "these will be base64 encoded into the json we return, and parsed on the way out.". I guess this is something that does not exist at application level and is invisible to the contract developer.

Reply:
Yeah, this is my comment how the raw json works.
If you parse it as binary data on the receiver it works fine. If you just eyeball the raw json, it looks a bit ugly. A note for later to think about, once all code is working well... maybe we can transparently upgrade the format.